README.MD

```diff
@@ -43,19 +43,33 @@ npm config set no-proxy localhost,127.0.0.1,example.org
 Additionally proxy settings found in the environment variables `HTTP_PROXY`, `HTTPS_PROXY` and `NO_PROXY` will be used if they are not defined in the `.npmrc` file.
 
 ### Cache Downloaded Binaries
-By default downloaded chromium binaries are not cached on the filesystem. You may enable caching by setting the `CHROMIUM_CACHE` environment variable to a directory path, for example:
+By default downloaded chromium binaries are cached in a local directory which varies by operating system:
+
+* Linux: `XDG_CACHE_HOME` or ~/.cache/node-chromium/
```

Comment:
Are you sure you want to define folders manually for each OS yourself? I think it might be a good idea to hold cache inside: `<your_project>/node_modules/.cache` (so, to have `.cache` folder used by npm). Like this, we don't care about access rights, people having incorrect system variables or smth else, if person can perform `npm install` then we should have all rights to save cache there (actually, 2 of our dependencies save theirs data inside `~/node-modules/.cache` so it looks like a standard practice to do so).

Reply:
Yeah fair question - I did a bit of research, there are all sorts of caching specifications and conventions. For example see this discussion: https://github.com/fatiando/pooch/issues/43

I should have mentioned, I basically copied what electron does: https://www.electronjs.org/docs/tutorial/installation#cache

Reply:
Yes, I do understand that it might be a solution for big libraries but it requires a lot of support from our side. It means that we are putting each OS and access rights support on ourselves. We need to check that these variables exist, that folders exist, that we got write access there and to do something else otherwise. And, if tomorrow, something changes in this architecture, we will have unexpected failures.
IMO, in our case our cache is a bit different from electron's cache as far as we are caching something related to `npm install` and it actually might reside inside `node_modules` and be shared with it (if someone decides to reuse `node_modules` folder).
What do you think?

Reply:
So it's your software and your decision, you can just call it how you want it and I'll make the change.

I think it *looks* more complex than it is.

* every option is in `homedir` of the current user and therefore we are justified in assuming we have write access
* the environment variables`LOCALAPPDATA` and `XDG_CACHE_HOME`  we can assume won't be set accidentally, if they are there we can trust them
* directories are created they don't exist
* if there are errors in caching we log the error to the user but keep going with the installation, so node_chromium will keep functioning regardless and the user can fix whatever issue is logged

So personally I like the idea of following standards for caching. On my machine when I run `ls ~/Library/Caches/` I get a **LOT** of directories listed. When I run `ls ~/.cache` there's about 3 directories.

For my use caching in `node_modules` would be no good because each job on the Jenkins agent has its own node_modules - without a shared cache every CI job will have to download the binary into its own cache using much more disk space and bandwidth.

Obviously it's your baby and you call.

Reply:
I see your point, Rick. Actually, you already did so much to this project that I think of it as yours as well, that's why I ask your opinion on what you think is the best. This project is done for the ease of using of everyone, so I don't want to put my own preferences inside but just make it the most useful for everyone.

I'm still thinking of access rights issues, because CI and web servers often got separate user with the least possible rights for execution / access. And, I think that system cache is a bit different purpose than ours module purpose.

I mean, you will always be able to change cache path if you wish but let me think a bit about the better default location.

Reply:
Hey I just had a thought - what about we don't "maintain" this problem here?
I could change it to use something like this: https://www.npmjs.com/package/cachedir ?

That way if standards change in future it's just a version bump in the package.json ...?

Reply:
@ricksbrown Yes, I absolutely agree that we don't need to do it ourselves and I prefer external package much more. Great, that you found a package that can help here.
Still, the question for me is if we store it in system or in project.
I have to think of possible use cases (multiple node installations using `nvm` / multiple projects on the same system / CI servers / ...). 

Thank you for your help, Rick. gn!